From 0c3941baef1928abcedd9d0de5ed34e7fa689b39 Mon Sep 17 00:00:00 2001
From: Hector Martin <hector@marcansoft.com>
Date: Mon, 7 Dec 2009 07:02:56 +0100
Subject: [PATCH] Get rid of SHORT_NOT_OK flag for USB sends

This flag is useful for reads only (and we don't want it on our reads
anyway). In DEBUG mode, the USB kernel driver is anal about this useless
flag and reports an error.
---
 daemon/usb-linux.c |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

diff --git a/daemon/usb-linux.c b/daemon/usb-linux.c
index 687e965..c334ef3 100644
--- a/daemon/usb-linux.c
+++ b/daemon/usb-linux.c
@@ -136,7 +136,6 @@ int usb_send(struct usb_device *dev, const unsigned char *buf, int length)
 	int res;
 	struct libusb_transfer *xfer = libusb_alloc_transfer(0);
 	libusb_fill_bulk_transfer(xfer, dev->dev, BULK_OUT, (void*)buf, length, tx_callback, dev, 0);
-	xfer->flags = LIBUSB_TRANSFER_SHORT_NOT_OK;
 	if((res = libusb_submit_transfer(xfer)) < 0) {
 		usbmuxd_log(LL_ERROR, "Failed to submit TX transfer %p len %d to device %d-%d: %d", buf, length, dev->bus, dev->address, res);
 		libusb_free_transfer(xfer);
@@ -149,7 +148,6 @@ int usb_send(struct usb_device *dev, const unsigned char *buf, int length)
 		xfer = libusb_alloc_transfer(0);
 		void *buffer = malloc(1);
 		libusb_fill_bulk_transfer(xfer, dev->dev, BULK_OUT, buffer, 0, tx_callback, dev, 0);
-		xfer->flags = LIBUSB_TRANSFER_SHORT_NOT_OK;
 		if((res = libusb_submit_transfer(xfer)) < 0) {
 			usbmuxd_log(LL_ERROR, "Failed to submit TX ZLP transfer to device %d-%d: %d", dev->bus, dev->address, res);
 			libusb_free_transfer(xfer);

 
